### DISCUSSION TITLE
Best practice for class bindings?

### DISCUSSION THREAD

**Discussion by HydrogenC** [AvaloniaUI/Avalonia#15148] [Q&A] [answered]
I'm working with a project in which, for xaml simplicity reasons, I packed some control properties into a class, like: 
```csharp
public class MyControlProperty : ObservableObject
{
    [ObservableProperty]
    int field1;

    [ObservableProperty]
    int field2;
}
```

And I redirected the property change of MyControlProperties onto the view model: 
```csharp
class MainViewModel : ViewModelBase
{
    public MainViewModel()
    {
        MyProperties.PropertyChanged += (sender, e) => OnPropertyChanged(nameof(MyProperties));
    }

    [ObservableProperty]
    MyControlProperty myProperties;
}
```

And in xaml I bind it to my own control: 
```xaml
<MyControl CustomProperty="{Binding MyProperties}" />
```

Then, when I change a field in MainViewModel.MyProperties.Field1, the OnPropertyChanged event of MainViewModel.MyProperties is indeed called as expected. But however, MyControl doesn't update.  
So why isn't the `OnPropertyChanged` event of `MyProperties` actually updating the bindings on `MyControl`? Is there any practise to address this?

**Comment by timunie** [comment]
the class should be public or at least internal.

**Reply by HydrogenC** [reply]
My bad. It is public, already.

**Comment by thevortexcloud** [comment, marked as answer]
Have you tried not raising the change event for `myProperties`? I am not 100% sure how Avalonia works internally but I suspect it might be unsubscribing from changes that `myProperties` itself raises. Semantically speaking: raising the event for `myProperties` means the property itself is changing, not the child properties.

**Reply by HydrogenC** [reply]
Yeah, that works! Thanks for your advice.
